mp4parse/src/lib.rs

```diff
@@ -1503,6 +1508,178 @@ fn read_infe<T: Read>(src: &mut BMFFBox<T>) -> Result<ItemInfoEntry> {
     Ok(ItemInfoEntry { item_id, item_type })
 }
 
+fn read_iref<T: Read>(src: &mut BMFFBox<T>) -> Result<TryVec<ItemReferenceEntry>> {
+    let mut entries = TryVec::new();
+    let version = read_fullbox_version_no_flags(src)?;
+    if version > 1 {
+        return Err(Error::Unsupported("iref version"));
+    }
+
+    let mut iter = src.box_iter();
+    while let Some(mut b) = iter.next_box()? {
+        let from_item_id = if version == 0 {
+            be_u16(&mut b)? as u32
+        } else {
+            be_u32(&mut b)?
+        };
+        let item_count = be_u16(&mut b)?;
+        for _ in 0..item_count {
+            let to_item_id = if version == 0 {
+                be_u16(&mut b)? as u32
+            } else {
+                be_u32(&mut b)?
+            };
+            entries.push(ItemReferenceEntry {
+                item_type: b.head.name.into(),
+                from_item_id,
+                to_item_id,
+            })?;
+        }
+    }
```

Comment:
It'd be nice to automate it, but it could be tackled as a separate refactoring.

Perhaps it could be automated by having BMFFBox keep a mutable reference to its BoxIter, and BMFFBox's `drop` could "poison" the iterator to return an error if the previous box hasn't been read fully.

Reply:
Ooh, that's an interesting idea. I've filed https://github.com/mozilla/mp4parse-rust/issues/246 for follow-up.